hail/src/main/scala/is/hail/expr/ir/InferType.scala

```diff
@@ -31,6 +34,11 @@ object InferType {
       case MakeStream(_, t, _) => t
       case MakeNDArray(data, shape, _, _) =>
         TNDArray(coerce[TIterable](data.typ).elementType, Nat(shape.typ.asInstanceOf[TTuple].size))
+      case StreamBufferedAggregate(_, _, newKey, _, _, aggSignatures) =>
+        val tupleFieldTypes = aggSignatures.map(_ => TBinary)
+        val tupleFields = (0 to tupleFieldTypes.length).zip(tupleFieldTypes).map { case (fieldIdx, fieldType) => TupleField(fieldIdx, fieldType) }.toIndexedSeq
+        val serializedAggSType = SStackStruct(TTuple(tupleFields), tupleFieldTypes.map(_ => EmitType(SBinaryPointer(PCanonicalBinary()), true)).toIndexedSeq)
```

Comment:
InferType shouldn't know about stypes. I think this can just be `TTuple(aggSignatures.map(_ => TBinary): _*)`